opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryMetricsExporter.java

```diff
@@ -0,0 +1,212 @@
+import io.opencensus.metrics.Metrics;
+import io.opencensus.metrics.export.Metric;
+import io.opencensus.metrics.export.MetricDescriptor;
+import io.opencensus.metrics.export.Point;
+import io.opencensus.metrics.export.Summary;
+import io.opencensus.metrics.export.Summary.Snapshot;
+import io.opencensus.metrics.export.TimeSeries;
+import io.opentelemetry.api.common.Labels;
+import io.opentelemetry.api.common.LabelsBuilder;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.DoublePoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.LongPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.SummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.ValueAtPercentile;
+import io.opentelemetry.sdk.resources.Resource;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class OpenTelemetryMetricsExporter extends MetricExporter {
+  private static final Logger LOGGER =
+      Logger.getLogger(OpenTelemetryMetricsExporter.class.getName());
+
+  private static final String EXPORTER_NAME = "OpenTelemetryMetricExporter";
+  private static final InstrumentationLibraryInfo INSTRUMENTATION_LIBRARY_INFO =
+      InstrumentationLibraryInfo.create("io.opentelemetry.opencensusshim", null);
+
+  private final IntervalMetricReader intervalMetricReader;
+  private final io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter;
+
+  public static OpenTelemetryMetricsExporter createAndRegister(
+      io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter) {
+    return new OpenTelemetryMetricsExporter(otelExporter, Duration.create(60, 0));
+  }
+
+  public static OpenTelemetryMetricsExporter createAndRegister(
+      io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter, Duration exportInterval) {
+    return new OpenTelemetryMetricsExporter(otelExporter, exportInterval);
+  }
+
+  private OpenTelemetryMetricsExporter(
+      io.opentelemetry.sdk.metrics.export.MetricExporter otelExporter, Duration exportInterval) {
+    this.otelExporter = otelExporter;
+    IntervalMetricReader.Options.Builder options = IntervalMetricReader.Options.builder();
+    MetricReader reader =
+        MetricReader.create(
+            MetricReader.Options.builder()
+                .setMetricProducerManager(Metrics.getExportComponent().getMetricProducerManager())
+                .setSpanName(EXPORTER_NAME)
+                .build());
+    intervalMetricReader =
+        IntervalMetricReader.create(
+            this, reader, options.setExportInterval(exportInterval).build());
+  }
+
+  @Override
+  public void export(Collection<Metric> metrics) {
+    ArrayList<MetricData> metricData = new ArrayList<>();
+    ArrayList<MetricDescriptor.Type> unsupportedTypes = new ArrayList<>();
```

Comment:
I think this should be a `HashSet`, otherwise it may contain duplicate types from every data point.
```suggestion
    Set<MetricDescriptor.Type> unsupportedTypes = new HashSet<>();
```